addons/web/static/src/views/fields/field.js

```diff
@@ -121,14 +124,47 @@ export class Field extends Component {
 
             const modifiers = fieldInfo.modifiers || {};
             readonlyFromModifiers = evalDomain(modifiers.readonly, evalContext);
-
             if (this.props.attrs) {
                 fieldInfo = {
                     ...fieldInfo,
                     attrs: { ...fieldInfo.attrs, ...this.props.attrs },
                 };
             }
-            propsFromNode = this.field.extractProps ? this.field.extractProps(fieldInfo) : {};
+            const dynamicInfo = {
+                get context() {
+                    const evalContext = record.getEvalContext
+                        ? record.getEvalContext(false)
+                        : record.evalContext;
+                    const context = omit(
+                        record.context,
+                        ...Object.keys(record.context).filter((key) => key.startsWith("default_"))
+                    );
+                    return {
+                        ...context,
+                        ...makeContext([fieldInfo.context], evalContext),
+                    };
+                },
+                readonly: readonlyFromModifiers,
+                get domain() {
+                    const evalContext = record.getEvalContext
+                        ? record.getEvalContext(true)
+                        : record.evalContext;
+
+                    return fieldInfo.domain
+                        ? new Domain(evaluateExpr(fieldInfo.domain, evalContext)).toList()
+                        : undefined;
+                },
+                get required() {
+                    return (
+                        fieldInfo.modifiers &&
+                        fieldInfo.modifiers.required &&
+                        evalDomain(fieldInfo.modifiers.required, record.evalContext)
+                    );
+                },
+            };
+            propsFromNode = this.field.extractProps
```

Comment:
suggestion:
if (this.field.extractProps) {
    const dynamicInfo = {...};
    propsFromNode = this.field.extractProps(fieldInfo, dynamicInfo);
}